_posts/en/newsletters/2024-04-10-newsletter.md

```diff
@@ -0,0 +1,331 @@
+  with the current public Bitcoin testnet (testnet3) and suggested
+  restarting it, potentially with a different set of special-case
+  consensus rules.
+
+  Previous versions of testnet had to be restarted when some people
+  began assigning economic value to testnet coins, resulting in them
+  becoming hard to acquire for free for people who wanted to perform
+  actual testing.  Lopp provided evidence of that happening again and
+  also described the well-known problem of block flooding due to
+  exploitation of testnet's custom difficulty adjustment algorithm.
+  Multiple people discussed potential changes to testnet to address that
+  and other problems, although at least one respondent [preferred][kim
+  testnet] allowing problems to continue as it made for interesting
+  testing.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review
+Club][] meeting, highlighting some of the important questions and
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[Implement 64 bit arithmetic op codes in the Script interpreter][review
+club 29221] is a PR by Chris Stewart (GitHub Christewart) that
+introduces new opcodes allowing users to perform arithmetic operations
+on larger (64-bit) operands in Bitcoin Script than is currently allowed
+(32-bit).
+
+This change, in combination with some existing soft fork proposals like
+[OP_TLUV][ml OP_TLUV] that enable transaction introspection, would allow
+users to build scripting logic based on a transaction's
+satoshi-denominated output values, which can easily overflow a 32-bit
+integer.
+
+Discussion on the approach, such as whether to upgrade existing opcodes
+or introduce new ones (e.g., `OP_ADD64`), is still ongoing.
+
+For more information, see the (WIP) [BIP][bip 64bit arithmetic], and the
+[discussion][delving 64bit arithmetic] on the Delving Bitcoin forum.
+
+{% include functions/details-list.md q0="What does the `CScriptNum`
+  `nMaxNumSize` parameter do?" a0="It represents the maximum size (in
+  bytes) of the `CScriptNum` stack element being evaluated. By default,
+  it is set to 4 bytes." a0link="https://bitcoincore.reviews/29221#l-34"
+
+  q1="What 2 opcodes accept 5-byte numeric inputs?"
+  a1="`OP_CHECKSEQUENCEVERIFY` and `OP_CHECKLOCKTIMEVERIFY` use signed
+  integers to represent time stamps. Using 4 bytes, that would put the
+  upper range of allowed dates in 2038. For that reason, a carve-out was
+  made for these 2 time-based opcodes to accept 5-byte inputs. This is
+  documented [in the code][docs 5byte carveout]."
+  a1link="https://bitcoincore.reviews/29221#l-45"
+
+  q2="Why was the `fRequireMinimal` flag introduced to `CScriptNum`?"
+  a2="`CScriptNum` has a variable length encoding. As described in
+  [BIP62][] (rule 4), this introduces opportunity for malleability. For
+  example, zero can be encoded as `[]`,<!-- skip-test --> `0x00`, `0x0000`, ... [Bitcoin
```

Comment:
Perhaps clearer:
```suggestion
  example, zero can be encoded as the empty vector,<!-- skip-test --> `0x00`, `0x0000`, ... [Bitcoin
```
Could also be avoided by choosing a different number.

Reply:
Changing to `OP_0` since that's a synonym for the empty byte array.